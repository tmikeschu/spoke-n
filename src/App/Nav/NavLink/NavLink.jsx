import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router'

const NavLink = props => (
  <Link {...props} activeClassName="active" />
)

NavLink.propTypes = {
  to: PropTypes.string.isRequired
}

export default NavLink
