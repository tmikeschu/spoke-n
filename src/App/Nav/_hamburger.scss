$height: 40px;
$stripe-height: $height * 0.05;
$height-transform: 11px;
$padding: 8px 0;
$width: $height + $padding;

.Hamburger {
  $root: &;
  border-radius: 5px;
  padding: $padding;
  height: $height;
  width: $height;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  box-sizing: border-box;
  &:hover {
    cursor: pointer;
  }

  div {
    background-color: $color2;
    height: $stripe-height;
    width: 100%;
    border-radius: 25px;
    transition: transform 500ms, opacity 500ms;
    animation: topBarLift 500ms;
  }

  #{$root}__first {
    animation: topBarLift 500ms;
  }

  #{$root}__third {
    animation: bottomBarDrop 500ms;
  }

  &--x {
    #{$root}__first {
      animation: topBarDrop 500ms;
      transform: translateY($height-transform) rotate(-45deg);
    }

    #{$root}__second {
      opacity: 0;
    }

    #{$root}__third {
      animation: bottomBarLift 500ms;
      transform: translateY(-($height-transform)) rotate(45deg);
    }
  }
}

@keyframes topBarDrop {
  from {
    transform: translateY(0);
  }
  50% {
    transform: translateY($height-transform);
  }
  to {
    transform: translateY($height-transform) rotate(-45deg);
  }
}

@keyframes topBarLift {
  from {
    transform: translateY($height-transform) rotate(-45deg);
  }
  50% {
    transform: translateY($height-transform);
  }

  to {
    transform: translateY(0);
  }
}

@keyframes bottomBarLift {
  from {
    transform: translateY(0);
  }
  50% {
    transform: translateY(-($height-transform));
  }
  to {
    transform: translateY(-($height-transform)) rotate(45deg);
  }
}

@keyframes bottomBarDrop {
  from {
    transform: translateY(-($height-transform)) rotate(45deg);
  }
  50% {
    transform: translateY(-($height-transform)) rotate(0);
  }
  to {
    transform: translateY(0);
  }
}
