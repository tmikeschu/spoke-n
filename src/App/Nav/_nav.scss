article.nav {
  background-color: $green;
  width: 30%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  ul {
    margin: 20px auto;
    display: inline-flex;
    width: 100%;
    justify-content: space-around;
    align-items: center;
    font-size: 1em;
    li {
      a {
        color: #fff;
        transition: 300ms;
        &:hover, &:focus, &.active {
          transition: 300ms;
          color: $yellow;
        }
      }
    }
  }
  @media (max-width: 640px) {
    width: 100%;
    flex-direction: row nowrap;
    ul {
      display: flex;
      flex-flow: row nowrap;
      margin: 0 auto;
      li {
        padding: 5px 0;
      }
    }
  }
}
