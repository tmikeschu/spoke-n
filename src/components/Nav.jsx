import React, { Component } from 'react';
import { Link } from 'react-router'
import '../stylesheets/App.css';

export default class Nav extends Component {
  render() {
    return (
      <article className="nav">
        <ul>
          <li><Link to="about">The Trip</Link></li>
          <li><Link to="#">Podcast Episodes</Link></li>
          <li><Link to="map">Map</Link></li>
          <li><Link to="#">Contact</Link></li>
        </ul>
      </article>
    );
  }
}
