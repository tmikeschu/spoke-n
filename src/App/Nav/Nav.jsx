import React from 'react'
import NavLink from './NavLink/NavLink'
import '../App.css'

const Nav = () => (
  <article className="nav">
    <ul>
      <li><NavLink to="about">Trip</NavLink></li>
      <li><NavLink to="map">Map</NavLink></li>
      <li><NavLink to="instagram">Photos</NavLink></li>
      <li><NavLink to="podcast">Podcast</NavLink></li>
      <li><NavLink to="contact">Contact</NavLink></li>
      <li><NavLink to="support">Support</NavLink></li>
    </ul>
  </article>
)

export default Nav

