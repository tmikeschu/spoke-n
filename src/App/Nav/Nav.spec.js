import React from 'react'
import { shallow } from "enzyme"
import Nav from "./Nav"
import NavLink from './NavLink/NavLink'

describe("<Nav />", () => {
  const nav = shallow(<Nav />)

  it("renders without crashing", () => {
    expect(nav).toBeTruthy()
  })

  it("renders the correct content", () => {
    expect(nav.equals(
      <article className="nav">
        <ul>
          <li><NavLink to="about">Trip</NavLink></li>
          <li><NavLink to="map">Map</NavLink></li>
          <li><NavLink to="instagram">Photos</NavLink></li>
          <li><NavLink to="podcast">Podcast</NavLink></li>
          <li><NavLink to="contact">Contact</NavLink></li>
          <li><NavLink to="support">Support</NavLink></li>
        </ul>
      </article>
    )).toBeTruthy()
  })

  describe("snapshot", () => {
    it("is valid", () => {
      expect(nav.getNodes()).toMatchSnapshot()
    })
  })
})
