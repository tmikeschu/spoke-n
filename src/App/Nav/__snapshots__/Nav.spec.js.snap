// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> snapshot is valid 1`] = `
Array [
  <article
    className="nav"
>
    <ul>
        <li>
            <NavLink
                to="about"
            >
                Trip
            </NavLink>
        </li>
        <li>
            <NavLink
                to="map"
            >
                Map
            </NavLink>
        </li>
        <li>
            <NavLink
                to="instagram"
            >
                Photos
            </NavLink>
        </li>
        <li>
            <NavLink
                to="podcast"
            >
                Podcast
            </NavLink>
        </li>
        <li>
            <NavLink
                to="contact"
            >
                Contact
            </NavLink>
        </li>
        <li>
            <NavLink
                to="support"
            >
                Support
            </NavLink>
        </li>
    </ul>
</article>,
]
`;
